packages/features/portfolio/src/containers/portfolio-state-zero/hooks/use-network-token-data.tsx

```diff
@@ -1,105 +1,67 @@
-import { TokenBalance, TokenDefinition } from '@notional-finance/core-entities';
-import { useCurrentNetworkStore } from '@notional-finance/notionable';
+import { usePortfolioStore } from '@notional-finance/notionable';
 import { PORTFOLIO_STATE_ZERO_OPTIONS } from '@notional-finance/util';
 
-type APYData = {
-  totalAPY?: number;
-};
-
-type ProductGroupItem = {
-  token: TokenDefinition;
-  apy: APYData;
-  tvl: TokenBalance;
-  liquidity: TokenBalance;
-  underlying?: TokenDefinition;
-  collateralFactor: string;
-};
-
-export type ProductGroupData = ProductGroupItem[][];
-
 export const getAPYDataForToken = (
-  symbol: string,
-  productGroupData: any[],
-  getHighestApy: boolean
+  activeToken: string,
+  productGroupData: any[]
 ) => {
   return productGroupData
-    .map((group) => {
-      const filteredGroup = group.filter((item) => {
-        return item?.underlying?.symbol === symbol;
-      });
-
-      if (filteredGroup.length === 0) return null;
-
-      return filteredGroup.reduce((prev, current) => {
-        const prevTotalAPY = prev.apy?.totalAPY || 0;
-        const currentTotalAPY = current.apy?.totalAPY || 0;
-        if (getHighestApy) {
-          return currentTotalAPY > prevTotalAPY ? current : prev;
-        } else {
-          return currentTotalAPY < prevTotalAPY ? current : prev;
-        }
-      });
+    .flatMap((group) => {
+      return group.filter((item) => item?.symbol === activeToken);
     })
     .filter((item) => item !== null);
 };
 
-export const getAvailableVaults = (
-  productGroupData: any[],
-  isPointsVault: boolean
-) => {
-  const productData = productGroupData.flat();
+export const getAvailableVaults = (productGroupData) => {
   const availableVaults: Set<string> = new Set();
-
-  const filterCondition = isPointsVault
-    ? (item: any) =>
-        item?.apy?.pointMultiples &&
-        item?.token?.tokenType === 'VaultShare' &&
-        item.apy?.totalAPY > 0
-    : (item: any) =>
-        !item?.apy?.pointMultiples &&
-        item?.token?.tokenType === 'VaultShare' &&
-        item.apy?.totalAPY > 0;
-
-  productData.filter(filterCondition).forEach((item) => {
-    if (item.underlying?.symbol) {
-      availableVaults.add(item.underlying.symbol);
-    }
-  });
-
+  productGroupData
+    .filter((item) => item.apy?.totalAPY > 0)
+    .forEach((item) => {
+      if (item?.symbol) {
+        availableVaults.add(item.symbol);
+      }
+    });
   return Array.from(availableVaults);
 };
 
-export const useNetworkTokenData = (
-  selectedTabIndex: number
-): {
-  tokenList: string[] | [];
-  productGroupData: ProductGroupData | [];
-  defaultSymbol: string;
-} => {
-  const currentNetworkStore = useCurrentNetworkStore();
+export const useNetworkTokenData = (selectedTabIndex: number) => {
```

Comment:
It doesn't look right to me that the set methods are not in a useEffect

Reply:
Imo this is an advantage not having them in a useEffect. This ensure's that the setters are only called once and then the data is just read.